From 36b86c22b9251ad17f8772404dd1fef40d7a9521 Mon Sep 17 00:00:00 2001
From: Michael Vassernis <michael.vassernis@tandemg.com>
Date: Mon, 2 May 2022 15:03:41 +0300
Subject: [PATCH 1/3] slac: fix cm_mnbc_sound_indicate not according to spec

	RND field size was incorrect (17 instead of 16).
	RND field should be preceded by 8 'Reserved' bytes.

Signed-off-by: Michael Vassernis <michael.vassernis@tandemg.com>
---
 docbook/slac.h.html | 3 ++-
 slac/slac.h         | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/docbook/slac.h.html b/docbook/slac.h.html
index 261d63de..592468e3 100644
--- a/docbook/slac.h.html
+++ b/docbook/slac.h.html
@@ -320,7 +320,8 @@ typedef struct __packed cm_mnbc_sound_indicate
 		uint8_t SenderID [SLAC_UNIQUE_ID_LEN]; 
 		uint8_t CNT; 
 		uint8_t RunID [SLAC_RUNID_LEN]; 
-		uint8_t RND [SLAC_UNIQUE_ID_LEN]; 
+		uint8_t RSVD [8]; 
+		uint8_t RND [SLAC_RND_LEN]; 
 	} 
 	MSVarField; 
 } 
diff --git a/slac/slac.h b/slac/slac.h
index 841cfd6a..71a05f0f 100644
--- a/slac/slac.h
+++ b/slac/slac.h
@@ -301,7 +301,8 @@ typedef struct __packed cm_mnbc_sound_indicate
 		uint8_t SenderID [SLAC_UNIQUE_ID_LEN];
 		uint8_t CNT;
 		uint8_t RunID [SLAC_RUNID_LEN];
-		uint8_t RND [SLAC_UNIQUE_ID_LEN];
+		uint8_t RSVD [8];
+		uint8_t RND [SLAC_RND_LEN];
 	}
 	MSVarField;
 }
-- 
2.25.1

